sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1148,6 +1148,14 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::GetCurlConnection(
         + std::string(curl_easy_strerror(result)));
   }
 
+  uint16_t port = request.GetUrl().GetPort();
+  if (port != 0 && !SetLibcurlOption(newHandle, CURLOPT_PORT, port, &result))
+  {
+    throw Azure::Core::Http::TransportException(
+        Details::DefaultFailedToGetNewConnectionTemplate + host + ". "
+        + std::string(curl_easy_strerror(result)));
+  }
```

Comment:
well, I don't think this would work.  Using CURLOPT_PORT works for using libcurl main flow. Since we use the manual protocol definition, we need to manually set the port as part of the url to open the connection (which I think we are already doing).   

It's ok to add this change, but I think it would have no effect at the end.  I remember doing tests on a local server at port XXXX and libcurl was working fine.

Reply:
Confirmed that this work as expected without setting curlopt port explicitly, but since there is no harm like you said, leaving it as is. This is good to know, thanks!